What is next? (#165)
LycheeOrg/LycheeOrg.github.io: src/content/post/2024-12-13-what-is-next.mdx

---
publishDate: 2024-12-13T12:00:00Z
author: ildyria
title: 'What is on the Santa Claus wish list?'
excerpt: We discuss the future of Lychee. What is the vision for the project? What is next on the roadmap?
image: https://images.unsplash.com/photo-1546188994-07c34f6e5e1b?q=80&w=3542&auto=format&fit=crop&ixlib=rb-4.0.3&ixid=M3wxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8fA%3D%3D
category: Release Notes
tags:
  - lychee
  - blabla
  - Supporter Edition
---

It has been a long time since last post: the [initial release of Version 6](/2024-10-26-v6).
Since then we have seen a few minor releases with the addition of a timeline view and a lot of bug fixing.
We still have a few more to resolve but I wanted to take a moment to talk about the future of Lychee.

At the moment we are only 2 in the team: [Martin Stone](https://github.com/d7415), the lead reviewer and [myself](https://github.com/ildyria), the main dev.

> As a result, due to being pretty much the sole developper, my vision for Lychee is pretty much the roadmap.

Reviewer: ```suggestion
As a result, due to being pretty much the sole developer, my vision for Lychee is pretty much the roadmap.
```

But let's get back to the subject. What is next for Lychee? What is the vision for the future of the project?

I see Lychee as a tool that can be used in multiple ways:
- as a **Gallery**: a self-hosted alternative of Flickr, Immich and a few others.
- as a **Pro Photographer**: a self-hosted alternative of pixieset.
- as a **Content Creator**: a self-hosted alternative of Patreon.

How do I plan to achieve this? Here is a list of things I would like to have in Lychee:

- `Content Creator`: have a feed similar to a Patreon page.  
  A post will be linked to an album. This will require the creation of a new page: *feed* where those will be displayed.
  However one notable difference with Patreon is that we do not plan to be multi-user support on that feature.
  In other words, the admin is the content creator.
- `Content Creator`: automatically create posts when a new albums is created.  
  As a potential continuation of the above, whenever a public album is created it could be included in the feed page.
  Note that those are just some ideas, nothing is set in stone.
- `Pro Photographer`: have a basket to "buy" pictures.  
  This will require a new page: *basket*.  

>   As a photographer I would like be able to sell my pictures, more easilly without having to deal with.

Reviewer: ```suggestion
  As a photographer I would like be able to sell my pictures, more easily without having to deal with.
```
Feels like a word or phrase is missing at the end here?

- `Automation`: possibility to upload an album as a Zip (a big part is already implemented, never merged).
- `Automation`: possibility to auto rename some files given some (user defined/toggled?) rules.
- `Automation`: set up a regular backup of the DB.
- `AI`: Facial/Person recognition.
  While Lychee aims to be privacy preserving, this is often a highly thought for feature.

In term of Quality of Life improvements that are needed in Lychee:

- `Must have`: User Groups.
- `Must have`: Permission propagations.
- `Must have`: Mass sharing v4 like in a single page.
- `Must have`: Refactoring settings to have ability to order category and options and most likely a new layout.
- `Mobile UI`: Allow multiple photo selection.
- `Tags`: Refactor the complete tag system to be more flexible (including tag albums).

With regard to security, I would like to see the following improvements:
- Photo storage at rest encrypted (this will need a big warning sign because if the key is lost bye bye data).
  However the implementation may require some smart ideas... E.g. where do we decrypt?
- Password locked albums even when logged in.

> But eventhough I there are still so many things to had, one thing that I really want to keep is the simplicity.

Reviewer: ```suggestion
But even though I there are still so many things to had, one thing that I really want to keep is the simplicity.
```
